### PR TITLE
Onl 6862 show anchor tags in side dropdown

### PR DESCRIPTION
**Pull request by DisegnAle**
**What kind of change does this PR introduce? (Bugfix, feature, docs update, ...)**
- added ec-btn dropdown props ('href', 'to')
- turned the button-dropdown's menu items and button to be a anchor tag (with a <a> tag or a router-link)
(connected to https://github.com/Ebury/eburyonline/pull/204)
 
**Does this PR introduce a breaking change?**
No
 
**Any dependencies that need to get resolved before merging?**
No
 
**Are there any other important/related changes/requirements when merging or deploying this PR?**
N/A
 
**ERI link or screenshots**
N/A

### CODE REVIEW

File: src/components/ec-btn-dropdown/ec-btn-dropdown.spec.js [hunk 1]
```diff
@@ -51,6 +51,23 @@ describe('EcBtnDropdown', () => {
     expect(wrapper.emitted('click').length).toBe(1);
   });
 
+  it("should render an anchor tag with a 'href' prop", async () => {
```
Comment (mcibique) [inline src/components/ec-btn-dropdown/ec-btn-dropdown.spec.js:54]: The description should match the assertions.

`it('should emit a click event when the CTA button is clicked')`

File: src/components/ec-btn-dropdown/ec-btn-dropdown.vue [hunk 2]
```diff
@@ -28,6 +30,23 @@
       @open="isOpen = true"
       @close="isOpen = false"
     >
+      <template #item="{ item }">
```
Comment (mcibique) [inline src/components/ec-btn-dropdown/ec-btn-dropdown.vue:33]: We need to test this template, mount a component with items that have one item disabled, one with `href`, one with `to` and one with `attrs`. We can pass an random ID in the `id` and a clickSpy in `click`, check the ID is rendered on the item, check that spy is called when item is clicked.

File: src/components/ec-btn-dropdown/ec-btn-dropdown.vue [hunk 3]
```diff
@@ -99,40 +124,62 @@ export default {
...
-
-      &:hover {
-        @apply tw-border-key-3;
-        @apply tw-bg-key-3;
-      }
-
-      &:disabled {
-        @apply tw-text-gray-5;
-      }
+.ec-btn-dropdown {
```
Comment (mcibique) [inline src/components/ec-btn-dropdown/ec-btn-dropdown.vue:127]: We should revert changes that have been done by the prettier. It's hard to tell here what has changed.

File: src/components/ec-btn-dropdown/__snapshots__/ec-btn-dropdown.spec.js.snap [hunk 4]
```diff
@@ -78,6 +78,52 @@ exports[`EcBtnDropdown should render as expected 1`] = `
 </div>
 `;
 
+exports[`EcBtnDropdown should render with a 'href' prop 1`] = `
+DOMWrapper {
```
[the remark this answers is not in the document]
Comment (DisegnAle) [reply src/components/ec-btn-dropdown/__snapshots__/ec-btn-dropdown.spec.js.snap]: corrected. i also inserted a test case when the item passed is disabled

File: src/components/ec-btn-dropdown/ec-btn-dropdown.spec.js (hunk 1 above)
Comment (DisegnAle) [reply src/components/ec-btn-dropdown/ec-btn-dropdown.spec.js]: corrected

Comment (DisegnAle) [reply src/components/ec-btn-dropdown/ec-btn-dropdown.spec.js]: corrected

File: src/components/ec-btn-dropdown/ec-btn-dropdown.vue (hunk 2 above)
Comment (DisegnAle) [reply src/components/ec-btn-dropdown/ec-btn-dropdown.vue]: corrected

**Review by mcibique — APPROVED** [review]

**Pull request merged**